subiquity/models/snaplist.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2015 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+import attr
+
+
+log = logging.getLogger("subiquity.models.snaplist")
+
+
+@attr.s(cmp=False)
+class SnapInfo:
+    name = attr.ib()
+    summary = attr.ib()
+    publisher = attr.ib()
+    description = attr.ib()
+    confinement = attr.ib()
+    channels = attr.ib(default=attr.Factory(list))
+
+
+@attr.s(cmp=False)
+class ChannelSnapInfo:
+    channel_name = attr.ib()
+    revision = attr.ib()
+    confinement = attr.ib()
+    version = attr.ib()
+    size = attr.ib()
+
+
+@attr.s(cmp=False)
+class SnapSelection:
+    channel = attr.ib()
+    is_classic = attr.ib()
+
+
+class SnapListModel:
+    """The overall model for subiquity."""
+
+    def __init__(self, common):
```

Comment:
We take common, shouldn't we hang on to it in the class?

Reply:
Ah it doesn't need to take this parameter any more.